video_xblock/static/js/video_xblock.js

```diff
@@ -35,12 +55,16 @@ function VideoXBlockStudentViewInit(runtime, element) {
    */
   function receiveMessage(event) {
     var origin = event.origin || event.originalEvent.origin; // For Chrome, the origin property is in the event.originalEvent object.
-    if (origin !== document.origin)
+    if (origin !== document.location.protocol + "//" + document.location.host)
       // Discard a message received from another domain
       return;
     if (event.data && event.data.action === 'save_state' &&
-        window.videoXBlockSaveHandlers[event.data.xblockUsageId]) {
-      saveState(window.videoXBlockSaveHandlers[event.data.xblockUsageId], event.data.state);
+        window.videoXBlockSaveStateHandlers[event.data.xblockUsageId]) {
+      saveState(window.videoXBlockSaveStateHandlers[event.data.xblockUsageId], event.data.state);
+    }
+    if (event.data && event.data.action === 'analytics' &&
+        window.videoXBlockSaveEventHandlers[event.data.xblockUsageId]) {
+      publishEvent(window.videoXBlockSaveEventHandlers[event.data.xblockUsageId], event.data.event_data)
```

Comment:
it can be unified so:
* change the way of storing handlers;
* unify data interfaces (event.data.info).

```js
var usageId = element.attributes['data-usage-id'].value;
window.videoXBlockState = window.videoXBlockState || {};
var handlers = window.videoXBlockState.handlers = window.videoXBlockState.handlers || {};
handlers = handlers || {
  save_state: {},
  analytics: {},
};

...
try {
  var url = handlers[event.data.action][event.data.xblockUsageId];
  if (url) {
    sendData(url, event.data.info);
  }
} catch(err) {}
```